feat: add info smart wallets (#215)
reown-com/reown-docs: docs/appkit/features/smart-accounts.mdx

<Button name="Try Demo" url="https://appkit-lab.reown.com/" />

### Deployment

Smart Accounts are deployed alongside the first transaction. Until deployment, a precalculated address, known as the counterfactual address, is displayed. Despite not being deployed, the account can still sign using 6492 signatures.

### Supported Networks

Smart Accounts are available on the following networks:

- Base Sepolia
- BSC (Binance Smart Chain)
- Fraximal
- Linea
- Mode
- Optimism
- Polygon
- Polygon Mumbai

> - Sepolia

Reviewer: Can we update this? It supports all EVM chains that are listed on Viem's package. You can refer to the doc page under "Features" -> "Email and Social Login" for the text. You can add the same text here as well

Reviewer 2: For smart accounts we need to use Pimlico's supported chains, not viem as we depend on premissionless for them!

<Button name="Try Demo" url="https://appkit-lab.reown.com/" />

### Deployment

Smart Accounts are deployed alongside the first transaction. Until deployment, a precalculated address, known as the counterfactual address, is displayed. Despite not being deployed, the account can still sign using 6492 signatures.

### Supported Networks

Smart Accounts are available on the following networks:

- Base Sepolia
- BSC (Binance Smart Chain)
- Fraximal
- Linea
- Mode
- Optimism
- Polygon
- Polygon Mumbai
- Sepolia

### User Eligibility

> Smart Accounts are exclusively available for users utilizing email or social logins (any Magic login)

Reviewer 2: ```suggestion
Smart Accounts are exclusively available for embedded wallet users (email and social login)
```

## FAQ

### What is a Smart Account?

> A Smart Account in improves the traditional account experience by replacing Externally Owned Accounts (EOAs) with a Smart Contract that follows the [ERC-4337 standard](https://eips.ethereum.org/EIPS/eip-4337). This opens up many use cases that were previously unavailable. 

Reviewer 2: remove in

- **Custom Logic:** Create custom transaction rules and workflows that align with personal or business requirements.

### How do I get a Smart Account?

Existing AppKit Universal Wallet Users will be given the option to upgrade their account to a smart account. Once you upgrade you will still be able to access your EOA and self-custody your account. 

New AppKit Universal Wallet Users will be given smart accounts by default when they login for the first time.

### Does it cost anything?

There is a small additional cost for activating your smart account. The activation fee is added to the first transaction and covers the network fees required for deploying the new smart contract onchain.

### Can I export my Smart Account?

No, you cannot export your Smart Account. The Smart Account (SA) is deployed by the EOA and is owned by the EOA. Your EOA account will always be exportable.
Also is good to know that SA don't have seedphrases.

### Can I rotate the signer ?

Yes, you can rotate the signer manage it elsewhere if they want too. We expect there to be more UIs for this wallet in the future including Safe-Wallet

> 

Reviewer 2: this is not currently possible I wouldn't include it

### Can I withdraw all my funds from my Smart Account?

Yes, you can withdraw all your funds from your Smart Account.

### What are account names?

> Smart account addresses start with ’0x’ followed by 42 characters, this is the unique address of your smart account on the network. ‘0x’ addresses like this are long, unwieldy and unmemorable. AppKit allows you to assign a more memorable name for your smart account using [ENS](https://ens.domains/). 

Reviewer 2: it's not ENS, but ather this:
https://docs.ens.domains/resolvers/ccip-read

we have a ccip read resolver for `reown.id` names and you can get one via appkit

> You can assign a name to your account and this will act as an alias for your account that can be shared publicly and provide a better user experience. AppKit account names are followed by the wcn.id. 

Reviewer 2: reown.id

### What can I do with my account name?

> As AppKit smart account addresses are the same across the supported networks, you only need one account name which can then be used across the networks. 

Reviewer 2: I'd note that given we are using pimlico we are limited to their supported networks 
https://docs.pimlico.io/infra/platform/supported-chains
